src/htm/ntypes/Value.cpp

```diff
@@ -610,10 +611,10 @@ static void escape_json(std::ostream &o, const std::string &s) {
       o << "\\t";
       break;
     default:
-      if ('\x00' <= *c && *c <= '\x1f') {
-        o << "\\u" << std::hex << std::setw(4) << std::setfill('0') << (int)*c;
+      if ('\x00' <= sc && sc <= '\x1f') {
```

Comment:
I think this logic is wrong...not the sign of c.
What this should be doing is converting non-printing and non-ASCII characters to hex.
So it should be
```
if ('\x20' < *c && *c >= '\x1f') {
```

Reply:
> if ('\x20' < *c && *c >= '\x1f')

is that correct? It's just a complicated way to `*c == '\x1f'`

Reply:
Oops. That is really wrong.  Dyslexia I guess.
What it should do is be true for 0 through \x1f and \x7f to \xff.
So that would be 
```
if ("\x1f" <= *c || *c >= "\x7f")
```
Do I have it right this time?

Reply:
No, I still have it wrong.  I looked up the standard for JSON (https://www.json.org/)
and it uses \u with 4 hex characters.  Let me figure out what this needs to be.

Reply:
> No, I still have it wrong. I looked up the standard for JSON

oh, pused the former fix for the correct ranges, handing over to you now :smiling_imp:

Reply:
It says "Any codepoint except " or \ or control characters."
So that means that it allows non-ASCII characters.  If we are using UTF8 which I assume we are then escaping multibyte characters individually could confuse it.
So that means just he control characters 0 - 0x1f  and the one character 0x7f

```
if (0x1f <= *c || *c == 0x7f)
```

Reply:
Ok, I messed it up to start with.  So let me try to fix it.